core/src/main/java/org/infinispan/xsite/irac/IracUtils.java

```diff
@@ -0,0 +1,102 @@
+package org.infinispan.xsite.irac;
+
+import static org.infinispan.metadata.impl.PrivateMetadata.getBuilder;
+
+import org.infinispan.commands.write.DataWriteCommand;
+import org.infinispan.commands.write.WriteCommand;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.container.versioning.irac.IracVersionGenerator;
+import org.infinispan.metadata.impl.IracMetadata;
+import org.infinispan.metadata.impl.PrivateMetadata;
+import org.infinispan.util.logging.Log;
+
+/**
+ * Utility methods related to IRAC (async Cross-Site Replication)
+ *
+ * @author Pedro Ruivo
+ * @since 11.0
+ */
+public final class IracUtils {
```

Comment:
Currently these methods are only ever used by interceptors that extend `AbstractIracLocalSiteInterceptor`. Does it make more sense for these to be simply be defined in `AbstractIracLocalSiteInterceptor` and `AbstractIracRemoteSiteInterceptor`? Or do you envisage these methods being required elsewhere longer term.

Reply:
it can be merged and I'll do it.